setup.py

```diff
@@ -100,6 +104,14 @@ def get_version():
 VERSION = get_version()
 macros.append(('PSUTIL_VERSION', int(VERSION.replace('.', ''))))
 
+PY36_PLUS = sys.version_info[:2] >= (3, 6)
+CP36_PLUS = PY36_PLUS and sys.implementation.name == "cpython"
```

Comment:
We can't use `CPYTHON  = sys.implementation.name == "cpython"` because [`sys.implementation`](https://docs.python.org/3/library/sys.html#sys.implementation) was introduced in Python 3.3 with [PEP 421](https://peps.python.org/pep-0421/).
I'll move the definitions below `PYPY = ...`.